File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L2/05/arraylebihdarisatu.ts
import { Scanner } from 'readline-sync';

function main(): void {
  const sc = new Scanner();

  console.log('Enter a 4 by 4 matrix row by row: ');
  const matrix: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));

  for (let x = 0; x < 4; x++) {
    for (let y = 0; y < 4; y++) {
      matrix[x][y] = sc.nextDouble();
    }
  }

  console.log(`Sum of the elements in the major diagonal is ${calculateDiagonal(matrix)}`);
}

function calculateDiagonal(matrix: number[][]): number {
  let total = 0;
  for (let i = 0; i < matrix.length; i++) {
    total += matrix[i][i];
  }
  return total;
}

main();

File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L2/05/arraylebihdarisatu.kt
package IR

import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    println("Enter a 4 by 4 matrix row by row: ")
    val matrix = Array(4) { DoubleArray(4) }

    for (x in 0 until 4) {
        for (y in 0 until 4) {
            matrix[x][y] = sc.nextDouble()
        }
    }

    println("Sum of the elements in the major diagonal is ${calculateDiagonal(matrix)}")
}

fun calculateDiagonal(matrix: Array<DoubleArray>): Double {
    var total = 0.0
    for (i in matrix.indices) {
        total += matrix[i][i]
    }
    return total
}

